android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/service/address/OemIdExtractorService.java

```diff
@@ -5,13 +5,18 @@
 public class OemIdExtractorService {
 
   private final OemIdExtractor extractorV1;
+  private final OemIdExtractor extractorV2;
 
-  public OemIdExtractorService(OemIdExtractor extractorV1) {
-
+  public OemIdExtractorService(OemIdExtractor extractorV1, OemIdExtractor extractorV2) {
     this.extractorV1 = extractorV1;
+    this.extractorV2 = extractorV2;
   }
 
-  String extractOemId(String packageName) {
+  public String extractOemId(String packageName) {
+    String oemId = extractorV2.extract(packageName);
```

Comment:
how do you know which extractor to give priority? In the class api is not clear 
I suggest to receive as constructor argument a list of sorted extractors and iterate them.